### PR TITLE
Bug 2060499: Replace vm object correctly into template

### PR DESCRIPTION
**Pull request by upalatucci**

### CODE REVIEW

File: src/views/catalog/customize/cd.ts [hunk 1]
```diff
@@ -70,11 +71,7 @@ export const addCDToTemplate = (
   template: V1Template,
   cdSource: V1beta1DataVolumeSpec,
 ): V1Template => {
-  return produce(template, (draftTemplate) => {
-    if (!cdSource) return draftTemplate;
-    let virtualMachine = getTemplateVirtualMachineObject(draftTemplate);
-    virtualMachine = addInstallationCDRom(virtualMachine, cdSource);
+  const virtualMachine = addInstallationCDRom(getTemplateVirtualMachineObject(template), cdSource);
```
Comment (hstastna) [inline src/views/catalog/customize/cd.ts:74]: I asked myself what if `!cdSource`, as the line
`if (!cdSource) return draftTemplate;`
was removed. And so I've checked `addInstallationCDRom`, and I think that in [this line](https://github.com/kubevirt-ui/kubevirt-plugin/blob/main/src/views/catalog/customize/cd.ts#L30) we should better make a change from
`cdSource.source?.http` to
`cdSource?.source?.http`
WDYT? Could you add this change to this PR to make sure nothing in there will break?

**Review by hstastna — COMMENTED** [review]
Check my comment below. Otherwise great work! 👍🏽

File: src/views/catalog/customize/cd.ts (hunk 1 above)
Comment (upalatucci) [reply src/views/catalog/customize/cd.ts:74]: Yes sure :-) thanks

**Pull request merged**